### PR TITLE
Added player attack animation

### PR DESCRIPTION
**Pull request by GianiStatie**

### CODE REVIEW

File: Assets/Scripts/Interfaces/IDamageable.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using UnityEngine;
+
+public interface IDamageable
+{
+    public float Health { set; get; }
+    public bool Targetable { set; get; }
+    public bool Invincible { set; get; }
```
Comment (Ratoone) [inline Assets/Scripts/Interfaces/IDamageable.cs:7]: I would make the setter private for these. You don't want to change them from outside

File: Assets/Scripts/Interfaces/IDamageable.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    public void OnHit(float damage, Vector2 knockback);
```
Comment (Ratoone) [inline Assets/Scripts/Interfaces/IDamageable.cs:8]: It's not clear what `knockback` represents. Is it the source of the damage, the direction, or what? I initially thought it was the knockback strength (which could be nice to add separately), but it doesn't fit the type.

File: Assets/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -2,16 +2,33 @@
...
 
     // Update is called once per frame
     void Update() {
+        if (weapon.isAttacking) {
+            movement = Vector2.zero;
+            animator.SetFloat("Speed", movement.sqrMagnitude);
+            return;
+        }
+
+        bool attacked = Input.GetButtonDown("Fire1");
```
Comment (Ratoone) [inline Assets/Scripts/Player/PlayerMovement.cs]: Unfortunate naming. You are not being attacked, but are attacking

File: Assets/Scripts/Utils/DamageableCharacter.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+public class DamageableCharacter : MonoBehaviour, IDamageable
+{
+    public float _health = 3;
+    public bool _targetable = true;
+    public bool _invincible = false;
+
+    public bool isAlive = true;
+    public bool disableSimulation = false;
+    public bool canTurnInvincible = false;
+    public float invincibilityTime = 0.25f;
```
Comment (Ratoone) [inline Assets/Scripts/Utils/DamageableCharacter.cs]: I'm being annoying now :)). Please make these private and `[SerializeField]`. You already have the public properties, and the underlying values should remain private.

File: Assets/Scripts/Interfaces/IDamageable.cs (hunk 2 above)
Comment (GianiStatie) [reply Assets/Scripts/Interfaces/IDamageable.cs:8]: it's a vector2 of direction + strength in which you're knocked back.

File: Assets/Scripts/Interfaces/IDamageable.cs (hunk 1 above)
Comment (GianiStatie) [reply Assets/Scripts/Interfaces/IDamageable.cs:7]: can't make them private and still have set, get apparently

File: Assets/Scripts/Utils/DamageableCharacter.cs (hunk 4 above)
Comment (GianiStatie) [reply Assets/Scripts/Utils/DamageableCharacter.cs]: what's the purpose of calling it [SerializedField]?
`Public fields are serializable by default, just like the areas you mark with SerializeField.`

File: Assets/Scripts/Interfaces/IDamageable.cs (hunk 1 above)
Comment (GianiStatie) [reply Assets/Scripts/Interfaces/IDamageable.cs:7]: it's a rule in interfaces apparently

File: Assets/Scripts/Utils/DamageableCharacter.cs (hunk 4 above)
Comment (Ratoone) [reply Assets/Scripts/Utils/DamageableCharacter.cs]: private setters. At first glance i dunno if i should be able to change them from outside of not

**Comment by Ratoone** [comment]
Fuck it, we're not a corporation :)). LGTM

**Pull request merged**